File: src/main/scala-graphql/io/github/graphql/scala/model/UserBlockedEventResponseProjection.scala
package io.github.graphql.scala.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection
import java.util.Objects

/**
 * Response projection for UserBlockedEvent
 */
@javax.annotation.Generated(
    value = Array("com.kobylynskyi.graphql.codegen.GraphQLCodegen"),
    date = "2021-03-07T19:49:36+0800"
)
class UserBlockedEventResponseProjection extends GraphQLResponseProjection {

    override def all$(): UserBlockedEventResponseProjection = all$(3)

    override def all$(maxDepth: Int): UserBlockedEventResponseProjection = {
        if (projectionDepthOnFields.getOrDefault("UserBlockedEventResponseProjection.ActorResponseProjection.actor", 0) <= maxDepth) {
            projectionDepthOnFields.put("UserBlockedEventResponseProjection.ActorResponseProjection.actor", projectionDepthOnFields.getOrDefault("UserBlockedEventResponseProjection.ActorResponseProjection.actor", 0) + 1)
            this.actor(new ActorResponseProjection().all$(maxDepth - projectionDepthOnFields.getOrDefault("UserBlockedEventResponseProjection.ActorResponseProjection.actor", 0)))
        }
        this.blockDuration()
        this.createdAt()
        this.id()
        if (projectionDepthOnFields.getOrDefault("UserBlockedEventResponseProjection.UserResponseProjection.subject", 0) <= maxDepth) {
            projectionDepthOnFields.put("UserBlockedEventResponseProjection.UserResponseProjection.subject", projectionDepthOnFields.getOrDefault("UserBlockedEventResponseProjection.UserResponseProjection.subject", 0) + 1)
            this.subject(new UserResponseProjection().all$(maxDepth - projectionDepthOnFields.getOrDefault("UserBlockedEventResponseProjection.UserResponseProjection.subject", 0)))
        }
        this.typename()
        this
    }

    def actor(subProjection: ActorResponseProjection): UserBlockedEventResponseProjection = {
        actor(null.asInstanceOf[String], subProjection)
    }

    def actor(alias: String, subProjection: ActorResponseProjection): UserBlockedEventResponseProjection = {
        fields.add(new GraphQLResponseField("actor").alias(alias).projection(subProjection))
        this
    }

    def blockDuration(): UserBlockedEventResponseProjection = {
        blockDuration(null.asInstanceOf[String])
    }

    def blockDuration(alias: String): UserBlockedEventResponseProjection = {
        fields.add(new GraphQLResponseField("blockDuration").alias(alias))
        this
    }

    def createdAt(): UserBlockedEventResponseProjection = {
        createdAt(null.asInstanceOf[String])
    }

    def createdAt(alias: String): UserBlockedEventResponseProjection = {
        fields.add(new GraphQLResponseField("createdAt").alias(alias))
        this
    }

    def id(): UserBlockedEventResponseProjection = {
        id(null.asInstanceOf[String])
    }

    def id(alias: String): UserBlockedEventResponseProjection = {
        fields.add(new GraphQLResponseField("id").alias(alias))
        this
    }

    def subject(subProjection: UserResponseProjection): UserBlockedEventResponseProjection = {
        subject(null.asInstanceOf[String], subProjection)
    }

    def subject(alias: String, subProjection: UserResponseProjection): UserBlockedEventResponseProjection = {
        fields.add(new GraphQLResponseField("subject").alias(alias).projection(subProjection))
        this
    }

    def typename(): UserBlockedEventResponseProjection = {
        typename(null.asInstanceOf[String])
    }

    def typename(alias: String): UserBlockedEventResponseProjection = {
        fields.add(new GraphQLResponseField("__typename").alias(alias))
        this
    }

    override def equals(obj: Any): Boolean = {
        if (this == obj) {
            return true
        }
        if (obj == null || getClass != obj.getClass) {
            return false
        }
        val that = obj.asInstanceOf[UserBlockedEventResponseProjection]
        Objects.equals(fields, that.fields)
    }

    override def hashCode(): Int = Objects.hash(fields)

}

File: src/main/java-graphql/io/github/graphql/model/UserBlockedEventResponseProjection.java
package io.github.graphql.model;

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField;
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection;

/**
 * Response projection for UserBlockedEvent
 */
@javax.annotation.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-03-06T19:47:47+0800"
)
public class UserBlockedEventResponseProjection extends GraphQLResponseProjection {

    public UserBlockedEventResponseProjection() {
    }

    @Override
    public UserBlockedEventResponseProjection all$() {
        return all$(3);
    }

    @Override
    public UserBlockedEventResponseProjection all$(int maxDepth) {
        if (projectionDepthOnFields.getOrDefault("UserBlockedEventResponseProjection.ActorResponseProjection.actor", 0) <= maxDepth) {
            projectionDepthOnFields.put("UserBlockedEventResponseProjection.ActorResponseProjection.actor", projectionDepthOnFields.getOrDefault("UserBlockedEventResponseProjection.ActorResponseProjection.actor", 0) + 1);
            this.actor(new ActorResponseProjection().all$(maxDepth - projectionDepthOnFields.getOrDefault("UserBlockedEventResponseProjection.ActorResponseProjection.actor", 0)));
        }
        this.blockDuration();
        this.createdAt();
        this.id();
        if (projectionDepthOnFields.getOrDefault("UserBlockedEventResponseProjection.UserResponseProjection.subject", 0) <= maxDepth) {
            projectionDepthOnFields.put("UserBlockedEventResponseProjection.UserResponseProjection.subject", projectionDepthOnFields.getOrDefault("UserBlockedEventResponseProjection.UserResponseProjection.subject", 0) + 1);
            this.subject(new UserResponseProjection().all$(maxDepth - projectionDepthOnFields.getOrDefault("UserBlockedEventResponseProjection.UserResponseProjection.subject", 0)));
        }
        this.typename();
        return this;
    }

    public UserBlockedEventResponseProjection actor(ActorResponseProjection subProjection) {
        return actor(null, subProjection);
    }

    public UserBlockedEventResponseProjection actor(String alias, ActorResponseProjection subProjection) {
        fields.add(new GraphQLResponseField("actor").alias(alias).projection(subProjection));
        return this;
    }

    public UserBlockedEventResponseProjection blockDuration() {
        return blockDuration(null);
    }

    public UserBlockedEventResponseProjection blockDuration(String alias) {
        fields.add(new GraphQLResponseField("blockDuration").alias(alias));
        return this;
    }

    public UserBlockedEventResponseProjection createdAt() {
        return createdAt(null);
    }

    public UserBlockedEventResponseProjection createdAt(String alias) {
        fields.add(new GraphQLResponseField("createdAt").alias(alias));
        return this;
    }

    public UserBlockedEventResponseProjection id() {
        return id(null);
    }

    public UserBlockedEventResponseProjection id(String alias) {
        fields.add(new GraphQLResponseField("id").alias(alias));
        return this;
    }

    public UserBlockedEventResponseProjection subject(UserResponseProjection subProjection) {
        return subject(null, subProjection);
    }

    public UserBlockedEventResponseProjection subject(String alias, UserResponseProjection subProjection) {
        fields.add(new GraphQLResponseField("subject").alias(alias).projection(subProjection));
        return this;
    }

    public UserBlockedEventResponseProjection typename() {
        return typename(null);
    }

    public UserBlockedEventResponseProjection typename(String alias) {
        fields.add(new GraphQLResponseField("__typename").alias(alias));
        return this;
    }


}
